### PR TITLE
feat: added types for Accordion, AccordionItem, AccordionSkeleton, SkeletonText

### LINKED ISSUE #13544: Add TypeScript types to Accordion, AccordionItem, AccordionSkeleton

### PR DESCRIPTION
**Pull request by SunnyJohal**
Closes #13544 #13576 

Added type definitions to the Accordion, AccordionItem, AccordionSkeleton, SkeletonTextComponents

#### Changelog

**New**

- Now using context instead of cloning the children within the accordion and accordion item.

**Changed**

- Added type definitions to the Accordion, AccordionItem, AccordionSkeleton, SkeletonTextComponents.

**Removed**

- Unused IconDescription prop.

#### Testing / Reviewing

Run unit tests, start up story book and confirm that the accordion is working as before.

### CODE REVIEW

**Comment by SunnyJohal** [comment]
@andreancardona I've created a new PR to get around the DCO issue, are you ok to review please. Thanks!

**Comment by andreancardona** [comment]
@SunnyJohal the only thing you need to do here is fix / update the tests so they are no longer failing - please let me know if I can help in any way!

**Comment by SunnyJohal** [comment]
Hi @andreancardona I've made the required changes yesterday but the ci still hasn't completed, is there anything we can do to get this over the line?

**Comment by andreancardona** [comment]
> Hi @andreancardona I've made the required changes yesterday but the ci still hasn't completed, is there anything we can do to get this over the line?

I just retriggered it - let's hope this passes!

**Comment by SunnyJohal** [comment]
@andreancardona the tests pass but now the percy stage is failing, not sure why this would be the case?

File: packages/react/src/components/Accordion/AccordionItem.tsx [hunk 1]
```diff
@@ -4,34 +4,128 @@
...
+   */
+  disabled?: boolean;
+
+  /**
+   * The handler of the massaged `click` event.
+   */
+  onClick?: MouseEventHandler<HTMLLIElement>;
+
+  /**
+   * he handler of the massaged `click` event on
```
Comment (andreancardona) [inline packages/react/src/components/Accordion/AccordionItem.tsx]:
```suggestion
   * The handler of the massaged `click` event on
```

**Review by andreancardona — APPROVED** [review]
🚀

File: packages/react/src/components/Accordion/AccordionItem.tsx [hunk 2]
```diff
@@ -4,34 +4,128 @@
...
-  disabled,
+  disabled: controlledDisabled,
+  handleAnimationEnd,
   ...rest
-}) {
+}: PropsWithChildren<AccordionItemProps>) {
   const [isOpen, setIsOpen] = useState(open);
   const [prevIsOpen, setPrevIsOpen] = useState(open);
   const [animation, setAnimation] = useState('');
+  const accodionState = useContext(AccordionContext);
```
Comment (tw15egan) [inline packages/react/src/components/Accordion/AccordionItem.tsx]:
```suggestion
  const accordionState = useContext(AccordionContext);
```

File: packages/react/src/components/Accordion/AccordionItem.tsx [hunk 3]
```diff
@@ -4,34 +4,128 @@
...
+
+  const disabledIsControlled = typeof controlledDisabled === 'boolean';
+  const disabled = disabledIsControlled
+    ? controlledDisabled
+    : accodionState.disabled;
```
Comment (tw15egan) [inline packages/react/src/components/Accordion/AccordionItem.tsx]:
```suggestion
    : accordionState.disabled;
```

File: packages/react/src/components/Accordion/Accordion.tsx [hunk 4]
```diff
@@ -8,7 +8,48 @@
...
+  size?: 'sm' | 'md' | 'lg';
+}
+
+interface AccordionV10Props extends Omit<AccordionProps, 'size'> {
+  /**
+   * Specify the size of the Accordion. Currently
+   * supports the following: `sm`, `md`, `lg`, `xl`.
+   */
+  size?: 'sm' | 'md' | 'lg' | 'xl';
+}
```
Comment (tw15egan) [inline packages/react/src/components/Accordion/Accordion.tsx]: Is there a reason we are bringing back the v10 `xl` size here?

File: packages/react/src/components/Accordion/AccordionItem.tsx (hunk 2 above)
Comment (SunnyJohal) [reply packages/react/src/components/Accordion/AccordionItem.tsx]: Thanks, changing that now

File: packages/react/src/components/Accordion/Accordion.tsx (hunk 4 above)
Comment (SunnyJohal) [reply packages/react/src/components/Accordion/Accordion.tsx]: I was just trying to be backwards compatible, but I will remove this

**Comment by SunnyJohal** [comment]
Hi @tw15egan I've made the requested changes and the CI is passing too, are you ok to approve so that we can merge this in please. Thanks!

**Review by tw15egan — APPROVED** [review]
Looks great, thanks for tackling this! 💪🏻 ✅

**Pull request merged**